src/storage/secondary/rowset/rowset_iterator.rs

```diff
@@ -265,14 +280,54 @@ impl RowSetIterator {
                     arrays[id] = Some(array);
                 }
             }
+            // For now, we only support range-filter scan by first column.
+            if id == 0 {
+                if !self.start_keys.is_empty() && !self.meet_start_key_before {
+                    // find the first row in range to begin with
+                    self.meet_start_key_before = true;
+                    let array = arrays[0].as_ref().unwrap();
+                    let len = array.len();
+                    let start_key = &self.start_keys[0];
+                    let start_row_id =
+                        (0..len).position(|idx| start_key - &array.get(idx) <= DataValue::Int32(0));
```

Comment:
Can do a binary search. As we will eventually remove this part, looks okay to me now.